### PR TITLE
chore(global_provider): Move methods to class as static

### PR DESCRIPTION
**Pull request by jfagoagas**
### Description

Improve code organisation making the `set_global_provider` and `get_global_provider` `@staticmethod`. This implies renaming a lot of code, which is already done and removes a file.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/common/models.py [hunk 1]
```diff
@@ -38,8 +38,8 @@ def __init__(self, arguments, bulk_checks_metadata):
         self.fixer = getattr(arguments, "fixer", None)
 
         # Shodan API Key
-        if arguments.shodan:
-            update_provider_config("shodan_api_key", arguments.shodan)
+        # if arguments.shodan:
```
Comment (jfagoagas) [inline prowler/providers/common/models.py]: Review this before merge.

File: prowler/config/config.py [hunk 2]
```diff
@@ -96,19 +95,6 @@ def check_current_version():
         return f"{prowler_version_string}"
 
 
-# TODO: revisit this function
```
Comment (jfagoagas) [inline prowler/config/config.py:99]: We cannot remove this function, it needs to be moved elsewhere.

**Review by pedrooot — DISMISSED** [review]
Thanks for your work! @jfagoagas 🥇

**Review by pedrooot — APPROVED** [review]
🔝

**Pull request merged**